state/upgrades.go

```diff
@@ -804,6 +804,38 @@ func AddStatusHistoryPruneSettings(pool *StatePool) error {
 	return nil
 }
 
+// AddStatusHistoryPruneSettings adds the model settings
+// to control log pruning if they are missing.
+func AddDefaultSpaceSetting(pool *StatePool) error {
+	st := pool.SystemState()
+	coll, closer := st.db().GetRawCollection(controllersC)
+	defer closer()
+	var doc settingsDoc
+	if err := coll.FindId(controllerSettingsGlobalKey).One(&doc); err != nil {
```

Comment:
This isn't the right collection. Take a look at _state/modelconfig.go_.
The collection is in `settingsC`. Model-global config is stored with a key of uuid:e (as in environ).

The method immediately below this one, `AddActionPruneSettings` uses the helper `applyToAllModelSettings`, which will be what you want to use to add this setting to all models.

Reply:
@manadart thanks for pointing out! Done